xarray/core/dataarray.py

```diff
@@ -1035,52 +1035,54 @@ def sel(
         return self._from_temp_dataset(ds)
 
     def head(
-        self, indexers: Mapping[Hashable, Any] = None, **indexers_kwargs: Any
+        self,
+        indexers: Union[Mapping[Hashable, Any], int] = None,
```

Comment:
While not actually use the default value of 5 in the method signatures, rather than setting it to None?

Reply:
I tried to set it there but it would raise `only one must be given` when providing only keyword arguments. This happened because whenever a call was made like this `ds.head(time=3)`, it would find the default value set for `indexers`, cast it into values of a dictionary, and then throw an error because `indexers` and `indexers_kwargs` are both set.  If I leave it to `None`, then I can easily avoid this kind of error by only setting the default value when no parameters are found.

The code below is basically the core of this PR
```python
if not indexers_kwargs:
    if indexers is None:
        indexers = 5
    if not isinstance(indexers, int) and not is_dict_like(indexers):
        raise TypeError("indexers must be a dict or a single integer")
if isinstance(indexers, int):
    indexers = {dim: indexers for dim in self.dims}
```

Reply:
Right, that makes sense. This seems to be about the best we can do then!